service/internal/configunmarshaler/dry_run_unmarshaller.go

```diff
@@ -0,0 +1,233 @@
+package configunmarshaler
+
+import (
+	"fmt"
+	"reflect"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type ExtensionValidateError struct {
+	Component string
+	Id        config.ComponentID
+	Err       string
+}
+
+type ExporterValidateError struct {
+	Component string
+	Id        config.ComponentID
+	Err       string
+}
+
+type ReceiverValidateError struct {
+	Component string
+	Id        config.ComponentID
+	Err       string
+}
+
+type ProcessorValidateError struct {
+	Component string
+	Id        config.ComponentID
+	Err       string
+}
+
+var ExtensionValidateErrors []ExtensionValidateError
+var ExportersValidateErrors []ExporterValidateError
+var ReceiversValidateErrors []ReceiverValidateError
+var ProcessorsValidateErrors []ProcessorValidateError
+
+func (ConfigUnmarshaler) DryRunUnmarshal(v *confmap.Conf, factories component.Factories) (*config.Config, error, bool) {
+	errorcheck := false
+	if err := v.Unmarshal(&rawCfg, confmap.WithErrorUnused()); err != nil {
```

Comment:
Using a var that is defined in an unrelated file would be confusing. If you have to use the same var, it might make sense to split this into its own source file.